Bottom: 590d78abca6568895476bf489103bd40362432d6
Top:    6c9fcd47e3c084ff437cc57bf9c6bae824f606fc
Author: Ian Kent <raven@themaw.net>
Date:   2013-06-07 13:54:07 +0800

autofs-5.0.7 - fix potential null dereference in lookup_mount()

Setting rv to CHE_OK by default could allow a null map entry dereference in
lookup_mount() in several modules.


---

diff --git a/modules/lookup_ldap.c b/modules/lookup_ldap.c
index 6e06e02..3371b3f 100644
--- a/modules/lookup_ldap.c
+++ b/modules/lookup_ldap.c
@@ -3012,7 +3012,7 @@ int lookup_mount(struct autofs_point *ap, const char *name, int name_len, void *
 				       mapent, ctxt->parse->context);
 	if (ret) {
 		time_t now = time(NULL);
-		int rv = CHE_OK;
+		int rv = CHE_FAIL;
 
 		/* Record the the mount fail in the cache */
 		cache_writelock(mc);
diff --git a/modules/lookup_sss.c b/modules/lookup_sss.c
index 5c2ed0a..013d3b4 100644
--- a/modules/lookup_sss.c
+++ b/modules/lookup_sss.c
@@ -673,7 +673,7 @@ int lookup_mount(struct autofs_point *ap, const char *name, int name_len, void *
 				       mapent, ctxt->parse->context);
 	if (ret) {
 		time_t now = time(NULL);
-		int rv = CHE_OK;
+		int rv = CHE_FAIL;
 
 		/* Record the the mount fail in the cache */
 		cache_writelock(mc);
